src/api/spec/controllers/search_controller_spec.rb

```diff
@@ -38,6 +39,21 @@
     subject! { get :package, params: { match: "@name='apacheX'" } }
 
     it_behaves_like 'find package'
+
+    context 'With offset' do
+      subject! { get :package, params: { offset: 'XXX', match: "@name='apacheX'" } }
+
+      it_behaves_like 'find package'
+    end
+  end
+
+  describe 'search for packages with offset = 1' do
+    let!(:package) { create(:package, name: 'apacheX', project: user.home_project) }
+    let!(:project) { create(:project_with_package, name: 'Foo', maintainer: user, package_name: 'apacheX') }
+
+    subject! { get :package, params: { limit: '1', offset: '1', match: "@name='apacheX'" } }
+
+    it_behaves_like 'find package'
```

Comment:
What about the other cases? Like no `limit` or no `offset`?

Reply:
it's being tested here https://github.com/openSUSE/open-build-service/blob/4dd82b05aad4342e1538d438ea56d3cd74b03328/src/api/spec/controllers/search_controller_spec.rb#L41

Reply:
Hm maybe I don't understand those parameters correctly. `limit=2` would mean out of 5 results I get 2. And `offset=4` would mean I get all results from result 5 onward right? Those cases I mean...

Reply:
you can see https://github.com/openSUSE/open-build-service/pull/11397#issuecomment-883320571 where i explain it (or at least try to) with examples... `offset=1` means we do like `results[1..]`. With `limit` you limit the number of results. So `offset=1` and `limit=4`, would mean something like `results[1..].take(4)` 

i.e:

```
2.6.5 :008 > offset=1
 => 1 
2.6.5 :009 > limit=4
 => 4 
2.6.5 :010 > 0.upto(10).to_a[offset..].take(limit)
 => [1, 2, 3, 4] 
2.6.5 :011 > 
```

Reply:
Okay, so where is this tested?

Reply:
We talked offline, this is in minitest and will be moved later...